src/apple/alohalytics_objc.mm

```diff
@@ -0,0 +1,354 @@
+    {"preferredLocalizations", preferredLocalizations},
+    {"localeIdentifier", ToStdString([locale objectForKey:NSLocaleIdentifier])},
+    {"calendarIdentifier", ToStdString([[locale objectForKey:NSLocaleCalendar] calendarIdentifier])},
+    {"localeMeasurementSystem", ToStdString([locale objectForKey:NSLocaleMeasurementSystem])},
+    {"localeDecimalSeparator", ToStdString([locale objectForKey:NSLocaleDecimalSeparator])},
+  };
+  if (device.systemVersion.floatValue >= 8.0) {
+    info.emplace("screenNativeBounds", RectToString(screen.nativeBounds));
+    info.emplace("screenNativeScale", std::to_string(screen.nativeScale));
+  }
+  Stats & instance = Stats::Instance();
+  instance.LogEvent("$iosDeviceInfo", info);
+
+  info.clear();
+  if (device.systemVersion.floatValue >= 6.0) {
+    if (device.identifierForVendor) {
+      info.emplace("identifierForVendor", ToStdString(device.identifierForVendor.UUIDString));
+    }
+    if (NSClassFromString(@"ASIdentifierManager")) {
+      ASIdentifierManager * manager = [ASIdentifierManager sharedManager];
+      info.emplace("isAdvertisingTrackingEnabled", manager.isAdvertisingTrackingEnabled ? "YES" : "NO");
+      if (manager.advertisingIdentifier) {
+        info.emplace("advertisingIdentifier", ToStdString(manager.advertisingIdentifier.UUIDString));
+      }
+    }
+  }
+  if (!info.empty()) {
+    instance.LogEvent("$iosDeviceIds", info);
+  }
+  // Force uploading to get first-time install information before uninstall.
+  instance.Upload();
+}
+
+// Returns <unique id, true if it's the very-first app launch>.
+static std::pair<std::string, bool> InstallationId() {
+  bool firstLaunch = false;
+  NSUserDefaults * userDataBase = [NSUserDefaults standardUserDefaults];
+  NSString * installationId = [userDataBase objectForKey:@"AlohalyticsInstallationId"];
+  if (installationId == nil) {
+    CFUUIDRef uuid = CFUUIDCreate(kCFAllocatorDefault);
+    // All iOS IDs start with I:
+    installationId = [@"I:" stringByAppendingString:(NSString *)CFBridgingRelease(CFUUIDCreateString(kCFAllocatorDefault, uuid))];
+    CFRelease(uuid);
+    [userDataBase setValue:installationId forKey:@"AlohalyticsInstallationId"];
+    [userDataBase synchronize];
+    firstLaunch = true;
+  }
+  return std::make_pair([installationId UTF8String], firstLaunch);
+}
+
+// Returns path to store statistics files.
+static std::string StoragePath() {
+  // Store files in special directory which is not backed up automatically.
+  NSArray * paths = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES);
+  NSString * directory = [[paths firstObject] stringByAppendingString:@"/Alohalytics/"];
+  NSFileManager * fm = [NSFileManager defaultManager];
+  if (![fm fileExistsAtPath:directory]) {
+    if (![fm createDirectoryAtPath:directory withIntermediateDirectories:YES attributes:nil error:nil]) {
+      NSLog(@"Alohalytics ERROR: Can't create directory %@.", directory);
```

Comment:
This is a critical error, isn't it? Do I read it right that you proceed with the initialization even if the directory could not have been created?